### PR TITLE
Issue #8651: Checkstyle fails on escape '\s' character

### LINKED ISSUE #8651: Checkstyle fails on escape '\s' character
Identified at https://github.com/checkstyle/checkstyle/pull/8634#discussion_r468594904
Check documentation: https://checkstyle.sourceforge.io/config_naming.html#ClassTypeParameterName

```
➜  IdeaProjects /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 jdk14/test/langtools/tools/javac/TextBlockLang.java
Note: jdk14/test/langtools/tools/javac/TextBlockLang.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  IdeaProjects cat single-module-config.xml                                                 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <!-- Here we use simple module to prove that we can parse files -->
    <module name="ClassTypeParameterName"/>
  </module>
</module>
➜  IdeaProjects cat jdk14/test/langtools/tools/javac/TextBlockLang.java
/*
 * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

/*
 * @test
 * @bug 8223967
 * @summary Unit tests for Text Block language changes
 * @compile --enable-preview -source ${jdk.version} -encoding utf8 TextBlockLang.java
 * @run main/othervm --enable-preview TextBlockLang
 */

public class TextBlockLang {
    public static void main(String... args) {
        test1();
        test2();
        test3();
    }

    /*
     * Test basic string functionality.
     */
    static void test1() {
        EQ("""
           """, "");
        EQ("""
            abc
            """, "abc\n");
        EQ("""

           """, "\n");
        EQ("""
            "
            """, "\"\n");
        EQ("""
            ""
            """, "\"\"\n");
        EQ("""
           \"""
           """, "\"\"\"\n");
        EQ("""
           "\""
           """, "\"\"\"\n");
        EQ("""
           ""\"
           """, "\"\"\"\n");
        EQ("""
            \r
            """, "\r\n");
        EQ("""
            \u2022
            """, "\u2022\n");
        EQ("""
            •
            """, "\u2022\n");
        LENGTH("""
            abc
            """, 4);
    }

    /*
     * Test escape-S.
     */
    static void test2() {
        if ('\s' != ' ') {
            throw new RuntimeException("Failed character escape-S");
        }
        EQ("\s", " ");
        EQ("""
           \s
           """, " \n");
    }

    /*
     * Test escape line terminator.
     */
    static void test3() {
        EQ("""
           abc \
           """, "abc ");
        EQ("\\\n".translateEscapes(), "");
        EQ("\\\r\n".translateEscapes(), "");
        EQ("\\\r".translateEscapes(), "");
    }

    /*
     * Raise an exception if the string is not the expected length.
     */
    static void LENGTH(String string, int length) {
        if (string == null || string.length() != length) {
            System.err.println("Failed LENGTH");
            System.err.println(string + " " + length);
            throw new RuntimeException("Failed LENGTH");
        }
    }

    /*
     * Raise an exception if the two input strings are not equal.
     */
    static void EQ(String input, String expected) {
        if (input == null || expected == null || !expected.equals(input)) {
            System.err.println("Failed EQ");
            System.err.println();
            System.err.println("Input:");
            System.err.println(input.replaceAll(" ", "."));
            System.err.println();
            System.err.println("Expected:");
            System.err.println(expected.replaceAll(" ", "."));
            throw new RuntimeException();
        }
    }
}
➜  IdeaProjects java -jar /home/nick/IdeaProjects/checkstyle/target/checkstyle-8.36-SNAPSHOT-all.jar -c single-module-config.xml jdk14/test/langtools/tools/javac/TextBlockLang.java

Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing jdk14/test/langtools/tools/javac/TextBlockLang.java
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:311)
	at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:221)
	at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:408)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:331)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:190)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:125)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: TokenStreamRecognitionException occurred while parsing file /home/nick/IdeaProjects/jdk14/test/langtools/tools/javac/TextBlockLang.java.
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:120)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:149)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:87)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:337)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:298)
	... 5 more
Caused by: line 84:15: unexpected char: 's'
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaLexer.nextToken(GeneratedJavaLexer.java:412)
	at antlr.TokenStreamHiddenTokenFilter.consume(TokenStreamHiddenTokenFilter.java:38)
	at antlr.TokenStreamHiddenTokenFilter.nextToken(TokenStreamHiddenTokenFilter.java:134)
	at antlr.TokenStreamSelector.nextToken(TokenStreamSelector.java:65)
	at antlr.TokenBuffer.fill(TokenBuffer.java:69)
	at antlr.TokenBuffer.LT(TokenBuffer.java:86)
	at antlr.LLkParser.LT(LLkParser.java:56)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.traditionalStatement(GeneratedJavaRecognizer.java:5653)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.statement(GeneratedJavaRecognizer.java:4815)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.compoundStatement(GeneratedJavaRecognizer.java:4544)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.field(GeneratedJavaRecognizer.java:3159)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.classBlock(GeneratedJavaRecognizer.java:3416)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.classDefinition(GeneratedJavaRecognizer.java:646)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.typeDefinitionInternal(GeneratedJavaRecognizer.java:561)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.typeDefinition(GeneratedJavaRecognizer.java:402)
	at com.puppycrawl.tools.checkstyle.grammar.GeneratedJavaRecognizer.compilationUnit(GeneratedJavaRecognizer.java:212)
	at com.puppycrawl.tools.checkstyle.JavaParser.parse(JavaParser.java:114)
	... 9 more
Checkstyle ends with 1 errors.

```

Checkstyle should be able to parse this file, since it is compilable.

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #8651: Checkstyle fails on escape '\s' character
This PR fixes #8651; all we needed to do was add the `\s` alternative to the `STD_ESC` lexer rule, so that the `STRING_LITERAL` rule would recognize this escape.  The text blocks grammar already supported `\s`, so no changes were necessary there. I have removed the two `BeforeExecutionExclusionFileFilter` modules specified [here](https://github.com/checkstyle/checkstyle/issues/8651#issuecomment-720490304) from ` openjdk14-excluded.files`.

Link to passing CI for the two `BeforeExecutionExclusionFileFilter` modules that have been removed: https://checkstyle.semaphoreci.com/jobs/465adbd8-5https://checkstyle.semaphoreci.com/jobs/465adbd8-57b9-4e87-8fc9-c3c883f2a9ef7b9-4e87-8fc9-c3c883f2a9ef

AST Regression test: https://nmancus1.github.io/diff_escapedS/diff/checkstyle/index.html

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java [hunk 1]
```diff
@@ -255,6 +256,13 @@ public void testInputJava14TextBlocksTabSize() throws Exception {
             getNonCompilablePath("java14/InputJava14TextBlocksTabSize.java"));
     }
 
+    @Issue("https://github.com/checkstyle/checkstyle/issues/8651")
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java:259]: I do not think this annotation is good.
we have git for this, and while we are strict on format for commit messages it is better to keep such information in git (hidden by default). Reader of code should focus on code first, any distractions(extra code lines) should be avoided.

Does it make sense to forbid it by Checkstyle validation ? as illegal class
the same as we forbid `@Test(expected = SomeException.class)` annotation of junit , will be enforce at https://github.com/checkstyle/checkstyle/issues/9137

**Review by romani — CHANGES_REQUESTED** [review]
items

File: src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java (hunk 1 above)
Comment (nmancus1) [reply src/test/java/com/puppycrawl/tools/checkstyle/grammar/AstRegressionTest.java:259]: https://github.com/checkstyle/checkstyle/issues/9138

**Comment by nmancus1** [comment]
Failing CI is from josm subversion repo connection error https://checkstyle.semaphoreci.com/jobs/b16db994-e870-4760-a75e-ab90390d3013

**Review by romani — APPROVED** [review]
ok to merge.

**Review by pbludov — APPROVED** [review]

**Pull request merged**